### PR TITLE
Type checking!

### PR DESCRIPTION
**Pull request by GJBoth**
This PR seriously upgrades our typing;
1. I've added the `NumberLike` and `ArrayLike` types.
2. Added pyright as type checker.
3. Added github CI for pyright, it passes!

For now it ignores the data folder as there's a lot do there, we can enable it later. 

@diptodip I addressed your comments so I think we're good, but could you triple check the merge didnt break any of your work?

### CODE REVIEW

**Comment by GJBoth** [comment]
Fixes #133

**Comment by diptodip** [comment]
Thanks! Probably all good but I'll take another look through.

File: src/chromatix/typing.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from jax import Array
+from numpy import ndarray
+
+ArrayLike = Array | ndarray
+NumberLike = ArrayLike | float | int
```
Comment (diptodip) [inline src/chromatix/typing.py:5]: Not important but I'm not sure I like this name, maybe `ScalarOrArray` or something? Is there any example in e.g. numpy or scipy we could look at?

Comment (diptodip) [reply src/chromatix/typing.py:5]: Also not sure if I like that we combine floats and ints together here but we can get to that later I guess.

File: src/chromatix/utils/initializers.py [hunk 2]
```diff
@@ -308,7 +310,7 @@ def ramp(center, theta_bounds, delta_ramp, ramp_defocus):
             grid[0] - jnp.sin(center) * defocus_center
         ) ** 2
         phase += ramp_mask * (ramp_defocus * ramp_quadratic)
-        phase -= ramp_mask * jnp.where(ramp_mask > 0, phase, 0).mean()
+        phase -= ramp_mask * jnp.where(ramp_mask > 0, phase, 0)  # typing: ignore
```
Comment (diptodip) [inline src/chromatix/utils/initializers.py:313]: We want to subtract the mean here

**Review by diptodip — CHANGES_REQUESTED** [review]

File: src/chromatix/utils/initializers.py (hunk 2 above)
Comment (GJBoth) [reply src/chromatix/utils/initializers.py:313]: Done!

File: src/chromatix/typing.py (hunk 1 above)
Comment (GJBoth) [reply src/chromatix/typing.py:5]: I changed it to ScalarLike.

**Review by diptodip — APPROVED** [review]

**Pull request merged**